### ISSUE TITLE
Doesn't seem to have support for Flags Enums

### ISSUE THREAD

**Issue by Cassidie** [ardalis/SmartEnum#59]
My team really likes this project and would like to use it for our codebase! Unfortunately, we have several flags enums and they don't seem to be compatible with this project. :(

The `FromValue` method throws a SmartEnumNotFoundException when given a flags enum value as well as casting a flags value.
```
// Given SomeEnum.A is 1 and SomeEnum.B is 2, all three throw the exception
SomeEnum.FromValue(SomeEnum.A | SomeEnum.B);
SomeEnum.FromValue(3); 
(SomeEnum)3
```
Also, there's no way to get the Name of multiple values with the logical OR operator like we get when the `Flags` attribute is placed on an enum (i.e. `SomeEnum.B | SomeEnum.C` => `B, C`). 

Will this project include this support in the near future?

**Comment by AFM-Horizon** [comment]
I'd like to have a go at this if that's ok.  I've started sketching up some ideas on an experimental branch as a solution.  If anyone has any specific ideas/constraints or suggestions please let me know!

**Comment by AFM-Horizon** [comment]
@Cassidie 
When you cast an int to a FlaggedEnum (SomeEnum)3 what behaviour would you normally expect from that? 
I'm a little confused as to how to implement this as to my mind this would return a list of 2 enum values right? SomeEnum.A and SomeEnum.B.
Cheers

**Comment by Cassidie** [comment]
@AFM-Horizon Yes, that's what I would expect as well. The two names, comma-separated, I believe is the default.

**Comment by AFM-Horizon** [comment]
@Cassidie Ok, cool.   Due to the smart enums often having extended behaviour attached to them I was thinking to return actual copies of the SmartEnum Objects.  and providing a ToString Type method to fulfill the more normal functionality if you just want the names, if that makes sense.  
I'll post up some usage examples/code snippets to demonstrate the functionality in the next few days.

**Comment by AFM-Horizon** [comment]
@Cassidie 
Hi! I've got something that looks more or less finished over on my master fork.  could you do me a favor and take a look at it to make sure it's behaving correctly? (To be honest I didn't even know what a flag enum was before I took on this issue haha) If you want to have a look through the README and the SmartFlagEnum Unit Tests that'd be awesome or feel free to fork it or clone it and run some tests on it if you like.  

Some notes:
- I have not yet implemented or found a way to allow casting to a List/Ienumerable from an int (SomeEnum)3

- I have written the algorithm to throw an exception if negative numbers other than -1 (representing All value) are provided as enum values.

- Negative numbers are still allowed to be passed in as a value, but do not function correctly due to their bit mask value being different and will give strange results, I was tossing up whether to throw an exception or not. thoughts?

- I have no checking logic to ensure that Power of two values are provided, and was thinking to have an optional safemode that throws an exception if these values are not provided as enum values at a minimum, thoughts?

Is this all making sense? 
thanks!

**Comment by AFM-Horizon** [comment]
@Cassidie I have put in a pull request, just to keep you in the loop.

**Comment by Cassidie** [comment]
First of all, great job taking on such a difficult task! I think you have the right ideas. I agree with the suggested exceptions, too. Thanks for the update!
